### PR TITLE
PLAT-1004: ensure the field names are stored using constant pointers …

### PR DESCRIPTION
**Pull request by wsourdeau**
…within StructureDescription

### CODE REVIEW

File: types/value_description.cc [hunk 1]
```diff
@@ -108,4 +108,49 @@ convertAndCopy(const void * from,
...
+
+/****************************************************************************/
+/* STRUCTURE DESCRIPTION BASE                                               */
+/****************************************************************************/
+
+void
+StructureDescriptionBase::
+operator = (const StructureDescriptionBase & other)
+{
+    ::fprintf(stderr, "StructureDescriptionBase copy %p from %p\n", this, &other);
```
Comment (jeremybarnes) [inline types/value_description.cc]: Debug output

**Comment by jeremybarnes** [comment]
This was already fixed in MLDB: https://github.com/mldbai/mldb/blob/master/types/value_description.cc#L360

I would consider taking the fix from MLDB to avoid the code diverging further.

**Comment by wsourdeau** [comment]
@jeremybarnes I thought so too, but the problem I am solving here does occur with MLDB (updated master and all) as well on Ubuntu 16.04. It has been occurring since at least 15.10. Although they solve a similar issue (hence their inclusion in this pr), the = operators do not change the nature of the problem because it occurs during the call to "addField/addFieldDesc" and not during the copying of instances.

**Comment by oktal** [comment]
+1

**Comment by wsourdeau** [comment]
Squash merged

**Pull request closed without merging**